src/java/com/LoginServlet.java

```diff
@@ -73,15 +63,52 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
     protected void doPost(HttpServletRequest request, HttpServletResponse response)
             throws ServletException, IOException {
         processRequest(request, response);
+
+        String message = null;
+
+        UserManagement loginUser = new UserManagement();
```

Comment:
Here you need to initialize the Jdbc class and give it the connection established by the listener class and store it at the session level. So basically:
```
HttpSession session = request.getSession(false);

Jdbc dbBean = new Jdbc();
dbBean.connect((Connection)request.getServletContext().getAttribute("connection"));
session.setAttribute("dbbean", dbBean);
```

I will change the `UserManagment` class to be static. So you wont to to initialize it so access its methods.
This means you will only need to interact with it in the following way:
```
User loggedInUser = UserManagement.loginUser(request.getParameter("username"), request.getParameter("password"), (Jdbc)session.getAttribute("dbbean"));
```
This will return an object which represents the user entity. If null is returned then their was a problem with the username and or password. To get the the user type just simply call `loggedInUser.getUserType()`. This can be used in your check from line 83 to 93. Tbh if you wanted to you could use a switch structure instead.